widgetastic_manageiq/__init__.py

```diff
@@ -1522,6 +1523,12 @@ def pages(self):
 
             # Adding 1 to pages_amount to include the last page in loop
             for page in range(1, self.pages_amount + 1):
+                # JSPaginatinoPane will wait for elements to load after changing page
+                # miq-tile-selection covers both - Grid and Tile view
+                if sel.is_displayed('//*[@class="miq-data-table"]'):
```

Comment:
maybe it could be more elegant. and container holding 'entities' could be stored. ideas?

Reply:
@lkhomenk, have you tried ensure_page_safe()?

btw, whether Karel is aware of this issue ? I haven't run into it when tested JS api.

Reply:
@izapolsk was pytest_selemiun counted as web_ui? @mfalesni  talked to sme DEVs, solution - add mechanism to check element is updated

Reply:
@lkhomenk, sure, this is part of it

Reply:
browser.plugin.ensure_page_safe doesn't help at all. paginator goes through all pages and entities can't have a chance to load. @izapolsk

Reply:
@lkhomenk pytest_selenium counts as web_ui and is definitely unwanted ... use `self.browser` instead.

Reply:
@izapolsk From what I have talked with Martin H. we will have less and less opportunities for JS checking as they move towards components.